expression/builtin_arithmetic_vec.go

```diff
@@ -105,11 +105,37 @@ func (b *builtinArithmeticMinusRealSig) vecEvalReal(input *chunk.Chunk, result *
 }
 
 func (b *builtinArithmeticMinusDecimalSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinArithmeticMinusDecimalSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	if err := b.args[0].VecEvalDecimal(b.ctx, input, result); err != nil {
+		return err
+	}
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDecimal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[1].VecEvalDecimal(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.MergeNulls(buf)
+	x := result.Decimals()
+	y := buf.Decimals()
+	to := new(types.MyDecimal)
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if err = types.DecimalSub(&x[i], &y[i], to); err != nil {
+			return err
+		}
+		x[i] = *to
```

Comment:
1. Is this ok to reuse `to` every time? The underlying data of `to` seems not reset before using.
If we want to reuse `to`, should we define a `Reset` member function for MyDecimal?
2. Will `x[i] = *to` cause a memory allocation? If so, can we exchange `x[i]` and `to` instead of allocating a new one every time?

Reply:
Please wait for this PR https://github.com/pingcap/tidb/pull/12600/files.